aspnet/6-pubsub/lib/BookDetailLookup.cs

```diff
@@ -223,15 +225,61 @@ public static void ProcessBook(IBookStore bookStore, long bookId)
             var response = WebRequest.Create(query).GetResponse();
             var reader = new StreamReader(response.GetResponseStream());
             var json = reader.ReadToEnd();
-            dynamic results = JObject.Parse(json);
-            dynamic info = results.items[0].volumeInfo;
-            book.Title = info.title;
-            book.PublishedDate = info.publishedDate;
-            book.Author = string.Join(", ", info.authors);
-            book.Description = info.description;
+            ParseBook(json, ref book);
             // TODO: update image.
             bookStore.Update(book);
         }
         // [END processbook]
+
+        /// <summary>
+        /// Parse a date time.  Return null if it can't be parsed.  A single number will be 
+        /// interpreted as a year, and the date returned will be YEAR-01-01.
+        /// </summary>
+        /// <param name="dateString">A string representation of the date.</param>
+        /// <returns>A datetime or null.</returns>
+        public static DateTime? ParseDate(string dateString)

```

Comment:
While reading `ParseBook`, I was frustrated to see another method, `ParseDate`, that I had to go find and I didn't understand what it's doing.

We know that the API will return one of 3 types of values for `volumeInfo.publishedDate`:
 - none (`null`)
 - year: `2000`
 - date: `2000-01-01`

So we don't need to `TryParse`.  The code can be more clear using `DateTime.parse()` or `new DateTime(year: month: date:)`

**FYI** `new DateTime().AddYears(2016- 1)` works too and will result in `1/1/2016 12:00:00 AM`.  The `- 1` is annoying and confusing however because `new DateTime().Year` is `1` and not `0`